### DISCUSSION TITLE
Camera 3 autofocus

### DISCUSSION THREAD

**Discussion by letkan** [Motion-Project/motionplus#162] [General]
The documentation is kind of sketchy on autofocus:
 ```
AfMode(int)

    AfModeManual = 0
    AfModeAuto = 1
    AfModeContinuous = 2
```
Does it mean AfMode(1) or AfMode=1 or AfModeAuto=1? 
Maybe this should be more like:
```
AfMode=0 (manual)
AfMode=1 (auto)
AfMode=2 (continuous)
```
After a lot of trial and error turns out autofocus as we know it, only works with AfMode=2, so what is AfMode=1 for? What does "auto" mean?

**Comment by Mr-Dave** [comment]
That would be something associated with the libcamera application.  The controls and descriptions in Motionplus are what was found in libcamera documentation as of the time of writing that code.  Since libcamera is a different application, the Motionplus documentation/log messages may just remove these messages/logs notices entirely going forward.   

Motionplus would instead just point users to review the [libcamera documentation](https://libcamera.org/api-html/namespacelibcamera_1_1controls.html) directly or search the [libcamera code](https://git.libcamera.org/libcamera/libcamera.git/tree/src/libcamera/control_ids.yaml?id=3e7f54a3128d8143afb200d626d3b208d493623a) for more details.

**Comment by letkan** [comment]
Indeed, just yesterday I came over libcamera docs and noticed that.
Thanks.
